sklearnex/svm/svc.py

```diff
@@ -232,38 +232,25 @@ def _onedal_gpu_supported(self, method_name, *data):
         patching_status = PatchingConditionsChain(
             f"sklearn.svm.{class_name}.{method_name}"
         )
+        self._is_sparse = sp.isspmatrix(data[0])
         if method_name == "fit":
-            if len(data) > 1:
-                self._class_count = len(np.unique(data[1]))
-            self._is_sparse = sp.isspmatrix(data[0])
-            patching_status.and_conditions(
-                [
-                    (
-                        self.kernel in ["linear", "rbf"],
-                        f'Kernel is "{self.kernel}" while '
-                        '"linear" and "rbf" are only supported on GPU.',
-                    ),
-                    (self.class_weight is None, "Class weight is not supported on GPU."),
-                    (
-                        self._class_count == 2,
-                        "Multiclassification is not supported on GPU.",
-                    ),
-                    (not self._is_sparse, "Sparse input is not supported on GPU."),
-                ]
-            )
-            return patching_status.get_status(logs=True)
+            self._class_count = len(np.unique(data[1]))
+        conditions = [
+            (
+                self.kernel in ["linear", "rbf"],
+                f'Kernel is "{self.kernel}" while '
+                '"linear" and "rbf" are only supported on GPU.',
+            ),
+            (self.class_weight is None, "Class weight is not supported on GPU."),
+            (not self._is_sparse, "Sparse input is not supported on GPU."),
+            (self._class_count == 2, "Multiclassification is not supported on GPU."),
+        ]
         if method_name in ["predict", "predict_proba", "decision_function"]:
-            patching_status.and_conditions(
-                [
-                    (
-                        hasattr(self, "_onedal_estimator")
-                        and self._onedal_gpu_supported("fit", *data),
-                        "oneDAL model was not trained on GPU.",
-                    )
-                ]
+            conditions.append(
+                (hasattr(self, "_onedal_estimator"), "oneDAL model was not trained")
             )
-            return patching_status.get_status(logs=True)
-        raise RuntimeError(f"Unknown method {method_name} in {class_name}")
```

Comment:
RuntimeError should exist at the end of function like in other estimators or this support checking workflow should be changed